source/linux/epoll_event_loop.c

```diff
@@ -288,6 +290,11 @@ static void s_schedule_task_future(struct aws_event_loop *event_loop, struct aws
     s_schedule_task_common(event_loop, task, run_at_nanos);
 }
 
+static void s_cancel_task(struct aws_event_loop *event_loop, struct aws_task *task) {
+    struct epoll_loop *epoll_loop = event_loop->impl_data;
+    aws_task_scheduler_cancel_task(&epoll_loop->scheduler, task);
```

Comment:
the task might be in `epoll_loop->task_pre_queue` 😢

Reply:
separate task